cardboardlint/linter_import.py

```diff
@@ -0,0 +1,43 @@
+import os
+import codecs
+from cardboardlint.common import Message, get_filenames
+
+
+def linter_import(config, files_lines):
+    """Linter for checking import statements.
+
+    Parameters
+    ----------
+    config : dict
+        Dictionary that contains the configuration for the linter
+    files_lines : dict
+        Dictionary of filename to the set of line numbers (that have been modified).
+        See `run_diff` method in `carboardlint`
+    """
+    messages = set()
+
+    # Find all (sub)package names, from which one should not import directly
+    packages = []
+    filenames = get_filenames(config['directories'], config['include'], config['exclude'],
+                              files_lines=files_lines)
+    for filename in filenames:
+        if filename.endswith('/__init__.py'):
+            packages.append(filename[:-12].replace('/', '.'))
+
+    # Loop all python and cython files
+    for filename in filenames:
+        # Only consider relevant files
+        if os.path.basename(filename).startswith('test_'):
+            continue
+        if filename.endswith('/__init__.py'):
+            continue
+        # Look for bad imports
+        with codecs.open(filename, encoding='utf-8') as f:
+            for lineno, line in enumerate(f):
+                for package in packages:
+                    # FIXME: i'm not too sure why all first import needs to have __version__
```

Comment:
I tried to make it easier to understand (e433cfd2b63668fb56000e3f3565aa00cd5a7784). It might still fail because `get_filenames` may return more than the bare minimum file path (like absolute path).